### ISSUE TITLE
E306 shouldn't warn when the command never fails

### ISSUE THREAD

**Issue by pikrzysztof** [ansible-community/ansible-lint#1250]
Why would I want pipefail to be set when:

A) the command can't fail because I said so
B) the failure condition is nontrivial, such as "the error message does not contain Y word"

?

Example code that E306 shouldn't warn about:

```
- shell: "pgrep stuff | wc -l" # noqa 306
  failed_when: false
  changed_when: false
  check_mode: false
  register: stuff_is_running
```

**Comment by ssbarnea** [comment]
I guess that the presence of `failed_when: false` should disable this rule. Feel free to make a PR to add this, it does make sense.
